src/algorithms/KirlikSayin.jl

```diff
@@ -0,0 +1,241 @@
+#  Copyright 2019, Oscar Dowson and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public License,
+#  v.2.0. If a copy of the MPL was not distributed with this file, You can
+#  obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+    KirlikSayin()
+
+`KirlikSayin` implements the Kirlik & Sayın (2014) algorithm.
+
+## Supported optimizer attributes
+
+"""
+
+mutable struct KirlikSayin <: AbstractAlgorithm
+  #yI::Union{Nothing, Vector{Float64}}
+  #yN::Union{Nothing, Vector{Float64}}
+  #L::Vector{Rectangle}
+
+  #KirlikSayin() = new(nothing, nothing, Rectangle[])
+  KirlikSayin() = new()
+end
+
+#=
+MOI.supports(::KirlikSayin, ::IdealPoint) = true
+
+function MOI.get(alg::KirlikSayin, attr::IdealPoint)
+  return alg.yI
+end
+
+function MOI.set(alg::KirlikSayin, attr::IdealPoint, value)
+  alg.yI = value
+  return
+end
+
+function MOI.set(alg::KirlikSayin, attr::IdealPoint, index, value)
+  alg.yI[index] = value
+  return
+end
+
+MOI.supports(::KirlikSayin, ::NadirPoint) = true
+
+function MOI.get(alg::KirlikSayin, attr::NadirPoint)
+  return alg.yN
+end
+
+function MOI.set(alg::KirlikSayin, attr::NadirPoint, value)
+  alg.yN = value
+  return
+end
+
+function MOI.set(alg::KirlikSayin, attr::NadirPoint, index, value)
+  alg.yN[index] = value
+  return
+end
+=#
+struct Rectangle
+  l::Vector{Float64}
+  u::Vector{Float64}
+  dim::Int
+  function Rectangle(l::Vector{Float64}, u::Vector{Float64})
+    @assert length(l) == length(u) "Dimension mismatch between l and u"
+    return new(l, u, length(l))
+  end
+end
+
+function ⊆(Rᵢ::Rectangle, Rⱼ::Rectangle)
+	@assert Rᵢ.dim == Rⱼ.dim "Dimension mismatch"
+	all(Rᵢ.l .≥ Rⱼ.l) && all(Rᵢ.u .≤ Rⱼ.u)
+end
+
+function remove_rect!(L::Vector{Rectangle}, R::Rectangle)
+  ix_to_pop = []
```

Comment:
```suggestion
    index_to_pop = Int[]
```
Two things: avoid things like `ix` because it makes it harder to read than `index`, and ensure all containers are type stable (if possible).